app/controllers/notes_controller.rb

```diff
@@ -5,20 +5,26 @@ def index
     else
       notes = Note.all
     end
-    render locals: { notes: notes }
+    perm_notes = []
+    notes.each do |note|
+      if current_permission?(note)
+        perm_notes << note
+      end
+    end
```

Comment:
Remember, whenever you are filling an array through an each, you could refactor that to use `.map`

```
notes.map do |note|
  note if current_permission?(note)
end
```